orchestration/src/main/java/com/sap/ai/sdk/orchestration/DpiMasking.java

```diff
@@ -75,11 +77,57 @@ public static class Builder {
     @Nonnull
     public DpiMasking withEntities(
         @Nonnull final DPIEntities entity, @Nonnull final DPIEntities... entities) {
-      val entitiesList = new ArrayList<DPIEntities>();
-      entitiesList.add(entity);
-      entitiesList.addAll(Arrays.asList(entities));
-      return new DpiMasking(maskingMethod, entitiesList, false, List.of());
+      val entitiesDTO =
+          Stream.concat(Stream.of(entity), Arrays.stream(entities))
+              .map(it -> (DPIEntityConfig) DPIStandardEntity.create().type(it))
+              .toList();
+      return new DpiMasking(maskingMethod, entitiesDTO, false, List.of());
     }
+
+    /**
+     * Specifies a custom regex pattern for masking.
+     *
+     * @param regex The regex pattern to match
+     * @param replacement The replacement string
+     * @return A new {@link DpiMasking} instance
+     */
+    @Nonnull
+    public DpiMasking withRegex(@Nonnull final String regex, @Nonnull final String replacement) {
+      val customEntity =
+          DPICustomEntity.create()
+              .regex(regex)
+              .replacementStrategy(
+                  DPIMethodConstant.create()
+                      .method(DPIMethodConstant.MethodEnum.CONSTANT)
+                      .value(replacement));
+
+      return new DpiMasking(maskingMethod, List.of(customEntity), false, List.of());
+    }
+  }
+
+  /**
+   * Specifies a custom regex pattern for masking.
+   *
+   * @param regex The regex pattern to match
+   * @param replacement The replacement string
+   * @return A new {@link DpiMasking} instance
+   */
+  @Nonnull
+  public DpiMasking withRegex(@Nonnull final String regex, @Nonnull final String replacement) {
```

Comment:
(MInor)

Please improve code quality of this method.
The amount of indentations has a code smell.